tracer/tracer.go

```diff
@@ -133,9 +133,22 @@ func (t *Tracer) Flush() error {
 		return nil
 	}
 
-	_, err := t.transport.Send(spans)
-	return err
+	// rebuild the traces list; this operation is done in the Flush() instead
+	// after each record() because this avoids a huge number of initializations
+	// and RW mutex locks, keeping the same performance as before (except for this
+	// little overhead). The overall optimization (and idiomatic code) could be
+	// reached replacing all our buffers with channels.
```

Comment:
in another PR we have to optimize the path to rebuild traces. We can easily remove this step if we refactor the code using channels. In the meantime I've added this step to prevent any changes to the current implementation because msgpack will be a big change and I don't want to introduce two big things to debug in case of errors